### ISSUE TITLE
Possible for beam x/y pols to incorrectly be compared to uvdata directional pols.

### ISSUE THREAD

**Issue by aewallwi** [HERA-Team/hera_sim#205]
The function  `_get_req_pols` currently assumes that the feed values for a uvbeam object with non None x_orientation are set to `N` or `E`. However, there are valid `uvbeam` objects where `feed_array` contains `'x'` or `'y'` with `x_orientation` specifying the mapping from the xy labels to the directions. As written, this function will incorrectly conclude that the `nn, `ee`, `ne` and `en` pols are not present in the uvbeam object (when converted to power).

https://github.com/HERA-Team/hera_sim/blob/4c0c324ebc99297c6eef742cbf93dcec880bf372/hera_sim/visibilities/vis_cpu.py#L414

**Comment by piyanatk** [comment]
@aewallwi Yes, I think you are correct that if `x_orientation` is `None`, they are assume to be `N` and `E`, but I might be wrong. We had lengthy discussion on how to implement this. I will leave it to @steven-murray or @r-pascua for details.

Can you provide example print out of `feed_array` and `x_orientation` of a UVBeam where the latter contain mapping? I think we should support that but it is a separate issue on default assumption of `x_orientation` (which I think we already have sensible a set up.)
